### PR TITLE
[Please test] More (minor) statistics cleanups / fixes

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Mostly cleanup of the statistics interface: Don't use global variables. Generate statistics where they're used. And a bug-fix: calculate statistics only for selected dives on HTML export. I didn't test the latter.

### Changes made:

See commit messages
### Related issues:

#1755 (contained in this PR)
### Additional information:

None.
### Release note:

Will do.
### Documentation change:

Not needed.
### Mentions:

@janmulder

### CODE REVIEW

**Comment by janmulder** [comment]
Quick try, mainly focused on the HTML generation. The HTML statistics now only (correctly) shows data from the selected dives. And the code cleanup looks sane. So ACK.

**Comment by bstoeger** [comment]
Rebased on master, added CHANGELOG.md entries for the last (potentially) user-visible changes.

File: core/statistics.h [hunk 1]
```diff
@@ -57,4 +62,25 @@ void selected_dives_gas_parts(volume_t *o2_tot, volume_t *he_tot);
...
+};
+inline stats_summary_auto_free::stats_summary_auto_free()
+{
+	init_stats_summary(this);
+}
+inline stats_summary_auto_free::~stats_summary_auto_free()
+{
+	free_stats_summary(this);
+}
+#endif
```
Comment (dirkhh) [inline core/statistics.h:83]: that's really clever. I learn something every day.

**Review by dirkhh — APPROVED** [review]
This looks like a potentially serious improvement.
And I learned something about C++ code and structures today :-)

**Pull request merged**